### ISSUE TITLE
Check BG every minute toggle not working on iPhone APP

### ISSUE THREAD

**Issue by Uksa007** [nightscout/nightguard#314]
Hi,

Using Nightguard iOS APP V3.29.2.906 (latest?) on iPhone

My Nightscout(web page) updates its data every one minute from the Libre 2 Sensor. 

When I turn on the "Check BG every minute" toggle in the iOS APP Preferences, it still only updates every 5 minutes, not every minute!

I have tried restarting the iOS APP.
Removing and reinstalling the iOS APP.

Is there anyway to get Nightguard to update every one minute?

**Comment by dhermanns** [comment]
@poml88 Any thoughts?

**Comment by poml88** [comment]
Not really. So, it is not working on the phone? The watch? Both? And Nightscout shows a new value every minute?
This is the official release from the app store?

Codewise, we just change the 5 for a 1, the rest remained the same... Maybe something with UserDefaults, does not remember the setting.

Btw, @dhermanns did you get my email? I sent you an email a while ago... :)

**Comment by Uksa007** [comment]
> Not really. So, it is not working on the phone? The watch? Both? And Nightscout shows a new value every minute? This is the official release from the app store?
> 
> Codewise, we just change the 5 for a 1, the rest remained the same... Maybe something with UserDefaults, does not remember the setting.

I'm testing on the iPhone. Don't have a watch to test. 
Using the official release from the app store.

Can you test, even if your sensor doesn't update every minute it should still check and update the time last checked beside the BG value, with the "Check BG every minute" toggle on?
MIne is always 5min before the BG is changing and the time last checked then goes back back to 1min.

Regards

**Comment by poml88** [comment]
Ok, thanks. I can test it on my setup, I have a Libre 3.

**Comment by poml88** [comment]
Ok, same here. Seems the phone app is doing something differently and ignoring the one minute option.
It did work on the watch so. I guess I never noticed, because I only cared about the watch.
However, I never tried the app store offical release. I only tried my own builds.

**Comment by Uksa007** [comment]
> Ok, same here. Seems the phone app is doing something differently and ignoring the one minute option

Thanks for testing and confirming that this is indeed a bug. 

How to progress fixing?
Who wrote the code for the Check BG every minute toggle, are they able to test and fix for the iPhone?

Regards

**Comment by Evs-nazgul** [comment]
I noticed the samenissue on my iwatch 7, a value is shown next to the BG, but the watch does not update every minute

**Comment by dhermanns** [comment]
@poml88 have you been able to achieve something here?

**Comment by poml88** [comment]
No,  not yet. It used to work great on my watch and since then no code changes. It is a bit strange. 
my own TestFlight is still valid, I can try to install that. Or maybe the user experiencing problems shall try to install that on their side, because that build worked great. 
just tested, works great on my testflight. I submitted it for external testing review.
Don’t know, it does not seem to work on your code tree, did you change something?

**Comment by dhermanns** [comment]
There are always changes, but not directly connected to the 1 minute updates. The whole project has been updated to iOS 18. That could alone be a reason for every kind of problems ;-).

I cannot guarantee the quality of that feature, since I don't have the hardware to test it in a real world environment.

You should use nightguard as your upstream project and take care of the feature.

**Comment by poml88** [comment]
Yes, sorry, well, maybe it is better to remove that feature. It was a contribution and I am happy to help, but the wind is quite hefty here in this project, which I guess is necessary to keep the thing kind of in order. 

That is part of why I started my own app. There I have a bit of creative freedom. I sent you an email about this but it seems you did not get it.

 Happy to look into this, but next week I have to cross the alps with the dog and will be far from any Xcode. 
 
I guess something went wrong with the PR. Have to find out what is missing or remove the thing.

**Comment by dhermanns** [comment]
Too bad - that was why I asked if you are able to support this feature since I can't. 

We will have to remove it if no one else in the community wants to assure the quality of this (for a longer time) :-/.

**Comment by poml88** [comment]
I look at the code. No idea. The PR was quite small, three files few lines. Those lines are still in the current code. 
It does work fine when I build it and in my testflight. It does not seem to work fine in your app store version.
I do not know why it is not working.

**Comment by dhermanns** [comment]
Great work Tomasz! Your change will be available with the next TestFlight. Let me know if this fixes the 1minute updates.

**Comment by dhermanns** [comment]
This time apple was fast. It is already available now with build 3.9.4.319:

https://testflight.apple.com/join/zE4CfcRn

**Comment by zibra** [comment]
Sorry for late reply. I've missed the notification. 
Tested it on my end and it works as expected now.
I've just noticed that the delta is showing 5 minutes change, which is probably best, but maybe it should be labeled in this case by something like `Δ5`...?

Video preview of 1min updates:
https://github.com/user-attachments/assets/6e402243-79d5-4c6e-b1d2-6397c05ee9bb

**Comment by dhermanns** [comment]
For 1min updates this is helpful, if you have missed readings or other problems receiving values from you sensor.

**Comment by Evs-nazgul** [comment]
Hello all,

Just informational,  I'm running version 3.9.4 on the test flight, and it ran fine until yesterday ( communciation issue with gluroo, which is fixed now), but I noticed this test flight case will end over 5 days.

The other OFFICAL version is 3.9.3.  Will this version be updated to 3.9.4, as this one ran smoothly sofar.

I reverted to the older 3.9.3 version now

**Comment by dhermanns** [comment]
Thanks for the hint. V3.9.4 will be released today.

**Comment by tast3rmonial** [comment]
[Cyber Monday Deal 30% OFF + A 35$ FREE GIFT Your Last Chance to Save](http=
s://tastermonial.com/products/freestyle-libre-3-sensors?variant=3D453889189=
15210)

[SHOP BEFORE IT ENDS](https://tastermonial.com/products/freestyle-libre-3-s=
ensors?variant=3D45388918915210)

Cyber Monday is your final call to take control of your health before the y=
ear ends.
Get 30% OFF your next order + a free $35 Baseline Experiment Box.

That=E2=80=99s a lot of value to finally start learning how your body react=
s to food,
stress, sleep, and sugar.

When Dec 2 hits, this deal disappears. No second round. Know your body
now or keep guessing.=E2=80=A8You choose.

[Wonderful product. Ease to order and obtain. Comfortable application of de=
vice. Overall wonderful device to learn what foods affect you but also can =
see how stress and poor sleep can affect numbers. I also liked the average =
glucose and seeing GMI. L.H. Skowhegan, ME](https://tastermonial.com/produc=
ts/freestyle-libre-3-sensors?variant=3D45388918915210)

[Food & Supplements](https://tastermonial.com/collections/food-and-suppleme=
nts-collection)

[Experiment Packages](https://tastermonial.com/collections/experimental-kit=
s)

[Smart Devices](https://tastermonial.com/collections/smart-devices)

[Support & Learning](https://help.tastermonial.com/en-US)

[instagram](www.instagram.com/tastermonial)
[youtube](www.youtube.com/channel/UCoaJvEFHR6RNBgv1CZBibTw/featured)

No longer want to receive these emails? [Unsubscribe](https://manage.kmail-=
lists.com/subscriptions/unsubscribe?a=3DTvyBXH&c=3D01JQXG2P1VW2MP7Y0V70VZ1T=
FY&k=3Df4021ecbcb64b616af0321da1faad71a&m=3D01KBD9RRZMJ4BDZKSKBPN6DF7G&r=3D=
01KBDG9G06D8DCJW32NVN5ZEXF).
TASTERMONIAL 20830 Stevens Creek Boulevard, #1063 Cupertino, California 950=
14

Tastermonial is a wellness brand built for people who want real answers fro=
m their body=E2=80=94not one-size-fits-all advice. We believe that no diet,=
 supplement, or routine should be based on guesswork. That=E2=80=99s why we=
=E2=80=99ve created tools that empower you to test how your own body respon=
ds to food, habits, and health products using real-time data. At the core o=
f Tastermonial is a simple idea: everyone=E2=80=99s body is different. What=
 spikes your glucose might be perfectly fine for someone else. What gives y=
ou stable energy might send someone else crashing. Instead of relying on ge=
neralized nutrition rules or fad diets, we give you the ability to test how=
 you respond=E2=80=94so you can build a routine that actually works for you=
r body. Our platform combines continuous glucose monitoring (CGM), guided t=
esting kits, and personalized protocols to help you uncover your metabolic =
patterns. With Tastermonial, you can monitor blood sugar reactions to speci=
fic meals, snacks, supplements, or even sleep and stress. It=E2=80=99s not =
about following someone else=E2=80=99s plan=E2=80=94it=E2=80=99s about usin=
g data to build your own. Tastermonial is trusted by health-conscious indiv=
iduals, biohackers, and wellness explorers who want to feel more in control=
 of their health journey. Whether you=E2=80=99re just getting started with =
glucose tracking or already knee-deep in optimizing your nutrition, our sys=
tem meets you where you are. We help you identify patterns, avoid crashes, =
improve energy levels, and make better decisions based on what=E2=80=99s re=
al=E2=80=94not what=E2=80=99s trending. Our test kits are curated to make e=
xperimenting easy. You=E2=80=99ll get specific food items, supplement proto=
cols, or meal combinations to track in a structured way. The experience is =
designed to be insightful without being overwhelming. By layering CGM data =
on top of real-life nutrition scenarios, you get clarity in just days=E2=80=
=94not months of trial and error. We believe that real health decisions sho=
uld be backed by personal data=E2=80=94not vague trends or influencer advic=
e. That=E2=80=99s why Tastermonial is built to help you explore your body=
=E2=80=99s unique reactions, take action based on real insight, and build s=
ustainable wellness habits that work in the real world. You don=E2=80=99t n=
eed another diet book. You need a feedback loop. That=E2=80=99s what Taster=
monial gives you. Our mission is to help people break free from generic nut=
rition rules and finally understand how to eat, move, and supplement in a w=
ay that aligns with their biology. We=E2=80=99re not just selling products=
=E2=80=94we=E2=80=99re building a smarter way to approach health. One that =
starts with curiosity, gets powered by data, and ends with better decisions=
 that improve how you feel every day. Join thousands of users who=E2=80=99v=
e used Tastermonial to uncover hidden glucose triggers, rethink their snack=
 choices, improve their energy, and feel more in control of their health. Y=
our body already has the answers. We just help you see them.
Tastermonial is a wellness brand built for people who want real answers fro=
m their body=E2=80=94not one-size-fits-all advice. We believe that no diet,=
 supplement, or routine should be based on guesswork. That=E2=80=99s why we=
=E2=80=99ve created tools that empower you to test how your own body respon=
ds to food, habits, and health products using real-time data. At the core o=
f Tastermonial is a simple idea: everyone=E2=80=99s body is different. What=
 spikes your glucose might be perfectly fine for someone else. What gives y=
ou stable energy might send someone else crashing. Instead of relying on ge=
neralized nutrition rules or fad diets, we give you the ability to test how=
 *you* respond=E2=80=94so you can build a routine that actually works for y=
our body. Our platform combines continuous glucose monitoring (CGM), guided=
 testing kits, and personalized protocols to help you uncover your metaboli=
c patterns. With Tastermonial, you can monitor blood sugar reactions to spe=
cific meals, snacks, supplements, or even sleep and stress. It=E2=80=99s no=
t about following someone else=E2=80=99s plan=E2=80=94it=E2=80=99s about us=
ing data to build your own. Tastermonial is trusted by health-conscious ind=
ividuals, biohackers, and wellness explorers who want to feel more in contr=
ol of their health journey. Whether you=E2=80=99re just getting started wit=
h glucose tracking or already knee-deep in optimizing your nutrition, our s=
ystem meets you where you are. We help you identify patterns, avoid crashes=
, improve energy levels, and make better decisions based on what=E2=80=99s =
real=E2=80=94not what=E2=80=99s trending. Our test kits are curated to make=
 experimenting easy. You=E2=80=99ll get specific food items, supplement pro=
tocols, or meal combinations to track in a structured way. The experience i=
s designed to be insightful without being overwhelming. By layering CGM dat=
a on top of real-life nutrition scenarios, you get clarity in just days=E2=
=80=94not months of trial and error. We believe that real health decisions =
should be backed by personal data=E2=80=94not vague trends or influencer ad=
vice. That=E2=80=99s why Tastermonial is built to help you explore your bod=
y=E2=80=99s unique reactions, take action based on real insight, and build =
sustainable wellness habits that work in the real world. You don=E2=80=99t =
need another diet book. You need a feedback loop. That=E2=80=99s what Taste=
rmonial gives you. Our mission is to help people break free from generic nu=
trition rules and finally understand how to eat, move, and supplement in a =
way that aligns with their biology. We=E2=80=99re not just selling products=
=E2=80=94we=E2=80=99re building a smarter way to approach health. One that =
starts with curiosity, gets powered by data, and ends with better decisions=
 that improve how you feel every day. Join thousands of users who=E2=80=99v=
e used Tastermonial to uncover hidden glucose triggers, rethink their snack=
 choices, improve their energy, and feel more in control of their health. Y=
our body already has the answers. We just help you see them.

**Comment by tast3rmonial** [comment]
Hey there,=C2=A0
=C2=A0
I wanted to personally reach out =E2=80=94 because today=E2=80=99s Giving T=
uesday, and we=E2=80=99re doing something a little different.
=C2=A0
Instead of asking for anything=E2=80=A6
We=E2=80=99re giving you something.
=C2=A0
30% off your next order
+ a free $35 Baseline Experiment Box
=C2=A0
No code. No stress. Just a simple thank you =E2=80=94 for being here, for c=
aring about your health, and for wanting to do something good for yourself =
(or someone you love).
=C2=A0
[Click here to claim your 30% off + free gift here >>](https://tastermonial=
.com/products/freestyle-libre-3-sensors?variant=3D45388918915210)
=C2=A0
If you=E2=80=99ve been on the fence, this is a good day to take that first =
step.
=C2=A0
The kit shows you exactly how your body responds to food, stress, sleep, an=
d sugar =E2=80=94 especially helpful through the holiday chaos.
=C2=A0
It=E2=80=99s not a detox. It=E2=80=99s not a diet.
It=E2=80=99s data. Real, personal, and powerful.
=C2=A0
This offer ends tonight.
But the impact can last way longer.

=C2=A0
All the best,
Tastermonial Customer Team

You received this email from TASTERMONIAL. If you would like to unsubscribe=
, [click here](https://manage.kmail-lists.com/subscriptions/unsubscribe?a=
=3DTvyBXH&c=3D01JQXG2P1VW2MP7Y0V70VZ1TFY&k=3Df4021ecbcb64b616af0321da1faad7=
1a&m=3D01KBG1T2PDCZAPT94N309932QA&r=3D01KBG2NPZCJKWPAW83DRJPJQVX).

**Comment by tast3rmonial** [comment]
Hey there,=C2=A0
=C2=A0
I wanted to personally reach out =E2=80=94 because today=E2=80=99s Giving T=
uesday, and we=E2=80=99re doing something a little different.
=C2=A0
Instead of asking for anything=E2=80=A6
We=E2=80=99re giving you something.
=C2=A0
30% off your next order
+ a free $35 Baseline Experiment Box
=C2=A0
No code. No stress. Just a simple thank you =E2=80=94 for being here, for c=
aring about your health, and for wanting to do something good for yourself =
(or someone you love).
=C2=A0
[Click here to claim your 30% off + free gift here >>](https://tastermonial=
.com/products/freestyle-libre-3-sensors?variant=3D45388918915210)
=C2=A0
If you=E2=80=99ve been on the fence, this is a good day to take that first =
step.
=C2=A0
The kit shows you exactly how your body responds to food, stress, sleep, an=
d sugar =E2=80=94 especially helpful through the holiday chaos.
=C2=A0
It=E2=80=99s not a detox. It=E2=80=99s not a diet.
It=E2=80=99s data. Real, personal, and powerful.
=C2=A0
This offer ends tonight.
But the impact can last way longer.

=C2=A0
All the best,
Tastermonial Customer Team

You received this email from TASTERMONIAL. If you would like to unsubscribe=
, [click here](https://manage.kmail-lists.com/subscriptions/unsubscribe?a=
=3DTvyBXH&c=3D01JV20E2HWY3G65DF3JXDTSKSV&k=3D7193b10ea2ca870547520568ca38dd=
27&m=3D01KBG1T2PDCZAPT94N309932QA&r=3D01KBG2NQ2K50ZGKVH968NVRQRC).

**Comment by tast3rmonial** [comment]
Hey there, 
 
I wanted to personally reach out — because today’s Giving Tuesday, and we’re doing something a little different.
 
Instead of asking for anything…
We’re giving you something.
 
30% off your next order
+ a free $35 Baseline Experiment Box
 
No code. No stress. Just a simple thank you — for being here, for caring about your health, and for wanting to do something good for yourself (or someone you love).
 
[Click here to claim your 30% off + free gift here >>](https://tastermonial.com/products/freestyle-libre-3-sensors?variant=45388918915210)
 
If you’ve been on the fence, this is a good day to take that first step.
 
The kit shows you exactly how your body responds to food, stress, sleep, and sugar — especially helpful through the holiday chaos.
 
It’s not a detox. It’s not a diet.
It’s data. Real, personal, and powerful.
 
This offer ends tonight.
But the impact can last way longer.

 
All the best,
Tastermonial Customer Team

You received this email from TASTERMONIAL. If you would like to unsubscribe, [click here](https://manage.kmail-lists.com/subscriptions/unsubscribe?a=TvyBXH&c=01JV22J2MKKZ7WGM3YVJ1RMQ3K&k=1af47cba633e30ac8a25523b2c360223&m=01KBG1T2PDCZAPT94N309932QA&r=01KBG2NPYK6E0RC2TKW5CQQBD2).
